cmd/pipeline/delete.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2018, Google, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package pipeline
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/spf13/cobra"
+	"github.com/spinnaker/spin/cmd/gateclient"
+	"github.com/spinnaker/spin/util"
+)
+
+type DeleteOptions struct {
+	*pipelineOptions
+	output      string
+	application string
+	name        string
+}
+
+var (
+	deletePipelineShort   = "Delete the provided pipeline"
+	deletePipelineLong    = "Delete the provided pipeline"
+	deletePipelineExample = `
+	usage: spin pipeline delete [options]
+
+	Delete the provided pipeline
+
+    --application: Spinnaker application the pipeline lives in
```

Comment:
Is there a way to generate the help text from the `PersistentFlags()`?

Reply:
```
# spin help
Global Options:

		--gate-endpoint               Gate (API server) endpoint.
		--no-color                    Removes color from CLI output.
		--insecure=false              Ignore certificate errors during connection to endpoints.
		--quiet=false                 Squelch non-essential output.
		--output <output format>      Formats CLI output.

Usage:
   [command]

Available Commands:
  application
  help        Help about any command
  pipeline

Flags:
      --config string          config file (default is $HOME/.spin/config)
      --gate-endpoint string   Gate (API server) endpoint. Default http://localhost:8084
  -h, --help                   help for this command
      --insecure               Ignore Certificate Errors
      --no-color               Disable color (default true)
      --output string          Configure output formatting
      --quiet                  Squelch non-essentual output

Use " [command] --help" for more information about a command.
```